app/controllers/user.js

```diff
@@ -23,3 +24,23 @@ exports.signUp = (req, res, next) =>
       logger.error(`Error creating user: ${err}`);
       return next(defaultError(err));
     });
+
+exports.signIn = (req, res, next) =>
+  getUserByEmail(req.body.email)
+    .catch(err => next(databaseError(err)))
```

Comment:
Specific errors like `databaseErrors` or `apiErrors` should be throw by services, in that way anywhere when is used error is already formatted

```
...some code
service1.action().catch(err => next(error1(err)))
.then(() => service2.action2().catch(err => next(error2(err))))
.then(...logic)
```
vs
```
...some code
service1.action()
.then(() => service2.action2()))
.then(...logic)
.catch(next)
```